pkg/cmd/cli/restore/delete.go

```diff
@@ -17,31 +17,190 @@ limitations under the License.
 
+	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
 
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	kubeerrs "k8s.io/apimachinery/pkg/util/errors"
+
+	arkv1api "github.com/heptio/ark/pkg/apis/ark/v1"
 	"github.com/heptio/ark/pkg/client"
 	"github.com/heptio/ark/pkg/cmd"
+	"github.com/heptio/ark/pkg/cmd/util/flag"
+	clientset "github.com/heptio/ark/pkg/generated/clientset/versioned"
 )
 
+// NewDeleteCommand creates and returns a new cobra command for deleting restores.
 func NewDeleteCommand(f client.Factory, use string) *cobra.Command {
+	o := &DeleteOptions{}
 	c := &cobra.Command{
-		Use:   fmt.Sprintf("%s NAME", use),
-		Short: "Delete a restore",
-		Args:  cobra.ExactArgs(1),
-		Run: func(c *cobra.Command, args []string) {
-			arkClient, err := f.Client()
-			cmd.CheckError(err)
+		Use:   fmt.Sprintf("%s [NAMES]", use),
+		Short: "Delete restores",
+		Example: `# delete a restore named "restore-1"
+	ark restore delete restore-1
+
+	# delete a restore named "restore-1" without prompting for confirmation
+	ark restore delete restore-1 --confirm
 
-			name := args[0]
+	# delete restores named "restore-1" and "restore-2"
+	ark restore delete restore-1 restore-2
 
-			err = arkClient.ArkV1().Restores(f.Namespace()).Delete(name, nil)
-			cmd.CheckError(err)
+	# delete all restores triggered by schedule "schedule-1"
+	ark restore delete --selector ark-schedule=schedule-1
+	
+	# delete all restores
+	ark restore delete --all`,
+
+		Run: func(c *cobra.Command, args []string) {
+			cmd.CheckError(o.Complete(f, args))
+			cmd.CheckError(o.Validate(c, f, args))
+			cmd.CheckError(o.Run())
 
-			fmt.Printf("Restore %q deleted\n", name)
 		},
 	}
-
+	o.BindFlags(c.Flags())
 	return c
 }
+
+// DeleteOptions contains parameters used for deleting a restore.
+type DeleteOptions struct {
+	Names     []string
+	All       bool
+	Selector  flag.LabelSelector
+	Confirm   bool
+	client    clientset.Interface
+	namespace string
+}
+
+// Complete fills in the correct values for all the options.
+func (o *DeleteOptions) Complete(f client.Factory, args []string) error {
+	o.namespace = f.Namespace()
+	client, err := f.Client()
+	if err != nil {
+		return err
+	}
+	o.client = client
+	o.Names = args
+	return nil
+}
+
+// Validate validates the fields of the DeleteOptions struct.
+func (o *DeleteOptions) Validate(c *cobra.Command, f client.Factory, args []string) error {
+	if o.client == nil {
+		return errors.New("Ark client is not set; unable to proceed")
+	}
+	var (
+		hasNames    = len(o.Names) > 0
+		hasAll      = o.All
+		hasSelector = o.Selector.LabelSelector != nil
+	)
+	if !xor(hasNames, hasAll, hasSelector) {
+		return errors.New("you must specify exactly one of: specific restore name(s), the --all flag, or the --selector flag")
+	}
+
+	return nil
+}
+
+// xor returns true if exactly one of the provided values is true,
+// or false otherwise.
+func xor(val bool, vals ...bool) bool {
```

Comment:
@carlisia, we should probably move this and `getConfirmation()` to a common place? Otherwise they'll be duplicated again for schedules too.

Reply:
+1 to this.  One option would be to create a `common/` package under `pkg/cmd/cli`.